tests/periph_rtt_min/main.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test to figure out RTT_MIN_OFFSET
+ *
+ * @author      Francisco Molina <francois-xavier.molina@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdatomic.h>
+
+#include "cpu.h"
+#include "periph_conf.h"
+#include "periph/rtt.h"
+
+#include "xtimer.h"
+
+/* min. amount of time to wait between set_alarm() */
+#define MIN_WAIT_US     (100)
+#define US_PER_TICK     (US_PER_SEC / RTT_FREQUENCY)
+
+#ifndef SAMPLES
+#define SAMPLES 1024LU
+#endif
+
+static atomic_bool cb_triggered;
+
+void cb(void *arg)
+{
+    (void)arg;
+    cb_triggered = true;
+}
+
+int main(void)
+{
+    uint32_t value = 0;
+
+    rtt_init();
+
+    printf("Evaluate RTT_MIN_OFFSET over %" PRIu32 " samples\n",
+           (uint32_t)SAMPLES);
+
+    for (unsigned i = 0; i < SAMPLES; i++) {
+        uint32_t offset = 0;
+        cb_triggered = false;
+        while (cb_triggered == false) {
```

Comment:
I'm trying to get this logic. basically it is setting an alarm, waiting the time it should take using xtimer, then verifying if the alarm triggered. IMO that's a bit brittle.
For synchronized RTTs, that'll probably work, but if the RTT works like a regular timer and just quantizes on the RTT clock, every set is late by `0<x<1` ticks, as the set_alarm() can happen anywhere within one RTT clock cycle.
This is supposed to figure out the smallest value that can be set without underflowing, right?
Would it maybe make sense to use the same logic as done for periph_timer_short_relative_set?
There, values are tried from larger to 0, until the timer clearly doesn't trigger anymore.

Reply:
In the case of `RTT` trying from larger values down might mean waiting for a long time, some of our RTT are running at `1Hz`. And why would going down from larger be better than going up? Seems just as brittle to me.

Reply:
ping @kaspar030!

Reply:
Ping @kaspar030?

Reply:
How about using `xtimer_mutex_lock_timeout()` instead?

Reply:
Did that, also found a bug in stm32f1 rtt code.

Reply:
Will split the stm32 change out.

Reply:
> I'm trying to get this logic. basically it is setting an alarm, waiting the time it should take using xtimer, then verifying if the alarm triggered. IMO that's a bit brittle.

It is waiting always longer then the time it should take to be set, I'm always waiting at least 100ul which is about 3 ticks for a 32Khz timer, maybe I can set that too ticks related.

Reply:
I added a 3ticks margin to the time waited.

Reply:
discussed offline, no issue here from my side anymore.